web/packages/teleport/src/Main/Main.tsx

```diff
@@ -169,13 +169,47 @@ export function Main(props: MainProps) {
   );
 }
 
-function renderRoutes(features: TeleportFeature[]) {
+function renderRoutes(
+  features: TeleportFeature[],
+  lockedFeatures: LockedFeatures
+) {
   const routes = [];
 
+  const lockedParents = getLockedParents(features, lockedFeatures);
+
   for (const [index, feature] of features.entries()) {
+    const isParentLocked = lockedParents.find(
+      parent => feature.parent && parent === feature.parent.name
+    );
+
+    // remove features with parents locked.
+    // The parent itself will be rendered if it has a lockedRoute,
+    // but the children shouldn't be.
+    if (isParentLocked) {
+      continue;
+    }
+
+    // add regular feature routes
     if (feature.route) {
       const { path, title, exact, component: Component } = feature.route;
+      routes.push(
+        <Route title={title} key={index} path={path} exact={exact}>
+          <CatchError>
+            <Suspense fallback={null}>
+              <Component />
+            </Suspense>
+          </CatchError>
+        </Route>
+      );
+    }
 
+    // add the route of the 'locked' variants of the features
+    if (feature.isLocked !== undefined && feature.isLocked(lockedFeatures)) {
+      if (!feature.lockedRoute) {
+        continue;
```

Comment:
It is "okay" as in no errors will occur (and the user will not see a feature they can't use anyway), but if this happens it is probably a developer error.

Reply:
That's right. It isn't happening because the only locked feature doesn't have a `route`, but it could happen.

Fixed in https://github.com/gravitational/teleport/pull/25588/commits/8aa01ce7e3a4ca58cf3ec97920784da9418d7857